### DISCUSSION TITLE
What is the general policy of using third-party crates in kernel development?

### DISCUSSION THREAD

**Discussion by ZhekaS** [tock/tock#4279] [Q&A] [answered]
Is there a policy dictating the usage of third-party crate dependencies when developing kernel (and perhaps the library as well)? I haven't seen much of external dependencies so far, so it looks like it is deprecated, or it is a wrong impression?

**Comment by brghena** [comment, marked as answer]
We actually have a whole policy document on this one! Take a look: https://github.com/tock/tock/blob/master/doc/ExternalDependencies.md I'm going to close this for now, but feel free to re-open with questions if you have any.

TLDR: we don't want to have external dependencies as much as possible, but there are some cases where it makes sense
